inbm-v5.md

```diff
@@ -0,0 +1,329 @@
+# In-band Manageability 5.0: Architecture
+
+## Overview
+
+In-band Manageability 5.0 (a.k.a. INBMv5, Turtle Creek v5) is a reimplementation of INBM in Golang.
+
+### Motivation
+
+`Re-architecting` and `Re-implementing` INBM is a major undertaking which was driven by the below listed motivation points:
+
+1. **Reduce complexity**: INBM primarily being a solution which is self-contained on a single compute device (Edge Node, IOT device etc.) was not leveraging `micro-services` architecture's inherent advantages but instead introduced additional complexity of managing and securing the services and their communication channels. With the re-architecture we plan on bringing all the `business-logic` of various agent within a single application/service thereby reducing complexity.
+1. **Improve performance**: Re-implementation of INBM will be done in `Golang` which is inherently better in performance w.r.t. Python being a compiled language as compared to interpreted.
+1. **Reduce footprint**: With all the functionality being brought into a single application the binary footprint overhead introduced by including common dependencies and Python interpreter in each agent will be removed.
+1. **Improve security and scalability**: Golang's characteristics of `statically typed`, `concurrency` and `memory management` helps building a more secure and optimized application.
+
+### Backward compatibility and features
+
+Like the earlier releases, the intension is to have as minimal an impact as possible for external consumers of INBM. This `backwards compatibility` requirement for INBMv5 insures that
+
+- the primary OTA feature set that INBM provided remain the same i.e.:
+  - OS Update
+  - Firmware Update
+  - Application update
+  - Basic telemetry and events reporting
+  - Device power control - reboot and shutdown
+
+- the primary `device-management` interfaces used and provided by INBM remain the same i.e.:
+  - `inbc`, command-line interface for local usage
+  - Azure IOT Central connectivity for `CSP` enablement
+  - ThingsBoard connectivity for `on-premise` device management
+
+> **NOTE** The availability of these features shall be staged in multiple releases starting with INBM v5.0
+
+## Architecture Diagram
+
+Below is a high-level architecture diagram for INBMv5 leveraging Golang's `multi-threading` capability and `channel` based inter-thread communication.
+
+![INBMv5 High-Level Architecture](./INBMv5-architecture.svg)
+
+   Figure 1: INBMv5 High-Level Architecture
+
+### Key Components
```

Comment:
Do we miss out inbm-diagnostics? Or it's features will be combined in inbm-daemon?

Reply:
I believe they will be combined in the inbm-daemon.